hyperspy/drawing/_widgets/rectangles.py

```diff
@@ -55,6 +60,24 @@ def _set_patch(self):
             picker=True,)]
         super(SquareWidget, self)._set_patch()
 
+    def _onjumpclick(self, event):
```

Comment:
I guess, this could be move to a more generic class, such as `Widget2DBase`? For 1D widget, it may be necessary to move in the child class and update accordingly.